pkgs/by-name/di/digikam/package.nix

```diff
@@ -1,149 +1,213 @@
+
+      kdePackages.kconfig
+      kdePackages.kxmlgui
+      kdePackages.ki18n
+      kdePackages.kwindowsystem
+      kdePackages.kservice
+      kdePackages.solid
+      kdePackages.kcoreaddons
+      kdePackages.knotifyconfig
+      kdePackages.knotifications
+      kdePackages.threadweaver
+      kdePackages.kiconthemes
+      kdePackages.kfilemetadata
+      kdePackages.kcalendarcore
+      kdePackages.kio
+      kdePackages.sonnet
+      # libksane and akonadi-contacts do not yet work when building for
+      # Qt 6.
+    ]
+    ++ lib.optionals (!stdenv.isDarwin) [
+      libGLX
+      libGLU
+    ]
+    ++ lib.optionals cudaSupport [ cudaPackages.cuda_cudart ];
+
+  checkInputs = [ qt6Packages.qtdeclarative ];
+
+  postConfigure = lib.optionalString finalAttrs.doCheck ''
+    ln -s ${testData} $cmakeDir/test-data
+  '';
 
   postPatch = ''
     substituteInPlace \
       core/dplugins/bqm/custom/userscript/userscript.cpp \
       core/utilities/import/backend/cameracontroller.cpp \
-      --replace-fail \"/bin/bash\" \"${lib.getExe bash}\"
+      --replace-fail '"/bin/bash"' ${lib.escapeShellArg "\"${runtimeShell}\""}
   '';
 
   cmakeFlags = [
-    "-DENABLE_MYSQLSUPPORT=1"
-    "-DENABLE_INTERNALMYSQL=1"
-    "-DENABLE_MEDIAPLAYER=1"
-    "-DENABLE_QWEBENGINE=on"
-    "-DENABLE_APPSTYLES=on"
-    "-DCMAKE_CXX_FLAGS=-I${libsForQt5.libksane}/include/KF5" # fix `#include <ksane_version.h>`
+    (lib.cmakeBool "BUILD_WITH_QT6" true)
+    (lib.cmakeBool "ENABLE_KFILEMETADATASUPPORT" true)
+    #(lib.cmakeBool "ENABLE_AKONADICONTACTSUPPORT" true)
+    (lib.cmakeBool "ENABLE_MEDIAPLAYER" true)
+    (lib.cmakeBool "ENABLE_APPSTYLES" true)
   ];
 
+  # Tests segfault for some reason…
+  # TODO: Get them working.
+  doCheck = false;
+
   dontWrapGApps = true;
 
   preFixup = ''
     qtWrapperArgs+=("''${gappsWrapperArgs[@]}")
-    qtWrapperArgs+=(--prefix PATH : ${lib.makeBinPath [ gnumake hugin enblend-enfuse ]})
-    qtWrapperArgs+=(--suffix DK_PLUGIN_PATH : ${placeholder "out"}/${libsForQt5.qtbase.qtPluginPrefix}/${pname})
+    qtWrapperArgs+=(--prefix PATH : ${
+      lib.makeBinPath [
+        gnumake
+        hugin
+        enblend-enfuse
+      ]
+    })
+    qtWrapperArgs+=(--suffix DK_PLUGIN_PATH : ${placeholder "out"}/${qt6Packages.qtbase.qtPluginPrefix}/digikam)
     substituteInPlace $out/bin/digitaglinktree \
       --replace "/usr/bin/perl" "${perl}/bin/perl" \
       --replace "/usr/bin/sqlite3" "${sqlite}/bin/sqlite3"
   '';
 
-  meta = with lib; {
-    description = "Photo Management Program";
-    license = licenses.gpl2;
-    homepage = "https://www.digikam.org";
-    maintainers = with maintainers; [ ];
-    platforms = platforms.linux;
+  meta = {
+    description = "Photo management application";
+    homepage = "https://www.digikam.org/";
+    changelog = "${finalAttrs.src.meta.homepage}-/blob/master/project/NEWS.${finalAttrs.version}";
+    sourceProvenance = [ lib.sourceTypes.fromSource ];
```

Comment:
```suggestion
```

https://nixos.org/manual/nixpkgs/unstable/#sec-meta-sourceProvenance

Reply:
Hmm, I see that this is technically a semantic no‐op according to the manual, but it’s intended to be a forward‐looking change: defining a `sourceProvenance` list conveys something different to using the default, because it implies a best-effort attempt at exhaustively outlining the provenance; hopefully, in the future, we can change the semantics so that an explicitly‐defined `sourceProvenance` is intended to convey something exhaustive, like `license` is, and at that point there would be a desire to annotate from‐source packages like this which I’m trying to get ahead of. Especially since there are definitely non‐source packages out there with no annotation.

Reply:
I'd prefer to get consensus before explicitly contradicting the manual.
And then the manual should be updated in that case.

Reply:
The manual doesn’t say that you shouldn’t define `sourceProvenance` like this. It just says that it currently doesn’t mean anything more than the empty string. But to humans, it clearly does, I think.

Do you mind if I ask on Matrix for opinions about this? If people are strongly against having this in there then I can remove it, but I do suspect we’re going to hopefully end up in a world where we expect everything to define `sourceProvenance` and I’d rather document it now than for there to be one more package that will need inspection and annotation later.

Reply:
> If a package contains elements that are not built from the original source by a nixpkgs derivation, the meta.sourceProvenance attribute should be a list containing one or more value from lib.sourceTypes defined in nixpkgs/lib/source-types.nix.

If we're expecting people to document this even for packages built from source, this bit needs to be changed.

Reply:
I wouldn’t say *expected*; I just don’t think it’s harmful in any way, and provides some benefit. To quote @toonn [on Matrix](https://matrix.to/#/!kjdutkOsheZdjqYmqp:nixos.org/$7tNoG-l590ubW99Vym0pJauotOcAds61hvZ9Ounr5lM?via=lossy.network&via=matrix.org&via=nixos.dev):

> I think it can only be helpful, not harmful. Don't even think the manual has to change, it says "if binary then SHOULD..." that says nothing about "if not binary," so it leaves it open to whatever the package maintainer feels like.
> Unless it's a lie of course, then it could be harmful.

As an analogy, consider licensing – we assume the best by default, so `license = lib.licenses.free;` is technically redundant, but it’s clearly more valuable to have that declaration than not, even if it has no strict semantic effect: it makes it clear that someone has thought about what licence the package is under and thinks it qualifies as FOSS. In the future, we may want to be more strict, requiring packages to list a `license` (perhaps with a treewide sweep for ones that don’t), and not assuming that packages with no licence declared are FOSS (as it has been wrong in the past); in that case, we’d be grateful for people who specified `license = lib.licenses.free;`. In my view, the exact same situation applies here, and I don’t think that’s in contradiction to what the manual says at all.

Reply:
To put it another way: If “from source” is assumed by default, and omission from the list is not a statement of absence, then why have `fromSource` at all, as it never conveys any additional information per the semantics given in the manual? Because it’s still providing more information that can be useful to readers now and policy in the future.

Reply:
> > If a package contains elements that are not built from the original source by a nixpkgs derivation, the meta.sourceProvenance attribute should be a list containing one or more value from lib.sourceTypes defined in nixpkgs/lib/source-types.nix.
> 
> If we're expecting people to document this even for packages built from source, this bit needs to be changed.

That reads, to me, like it's just explaining how you'd define things that are not built from source and that it may be multiple source types. Not, that you shouldn't use it when it is built from source.